### PR TITLE
implement connectablePosition modification for branch form

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java [hunk 1]
```diff
@@ -53,19 +52,89 @@ public class BranchModificationEntity extends BasicEquipmentModificationEntity {
...
+    })
+    private EnumModificationEmbedded<ConnectablePosition.Direction> connectionDirection2;
+
     @Embedded
     @AttributeOverrides(value = {
         @AttributeOverride(name = "value", column = @Column(name = "connected1")),
         @AttributeOverride(name = "opType", column = @Column(name = "connected1Op"))
     })
-    private BooleanModificationEmbedded connected1;
+    private BooleanModificationEmbedded terminal1Connected;
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java:130]: change the AttributeOverride column name

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java [hunk 2]
```diff
@@ -53,19 +52,89 @@ public class BranchModificationEntity extends BasicEquipmentModificationEntity {
...
 
     @Embedded
     @AttributeOverrides(value = {
         @AttributeOverride(name = "value", column = @Column(name = "connected2")),
         @AttributeOverride(name = "opType", column = @Column(name = "connected2Op"))
     })
-    private BooleanModificationEmbedded connected2;
+    private BooleanModificationEmbedded terminal2Connected;
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java:137]: same here

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 3]
```diff
@@ -54,6 +54,7 @@ public final class ModificationUtils {
     public static final String CONNECTION_NAME_FIELD_NAME = "Connection name";
     public static final String CONNECTION_DIRECTION_FIELD_NAME = "Connection direction";
     public static final String CONNECTION_POSITION_FIELD_NAME = "Connection position";
+    public static final String NETWORK_NOT_EXIST = " does not exist in network";
```
Comment (EtienneLt) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:57]: well this is not really that, maybe more like NOT_EXIST_IN_NETWORK

File: src/main/resources/db/changelog/changesets/changelog_20240904T211456Z.xml [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    <changeSet author="rehiligha (generated)" id="1725484515185-67">
+        <addColumn tableName="line_modification">
+            <column name="voltage_level_id2op" type="varchar(255)"/>
+        </addColumn>
+    </changeSet>
+    <changeSet author="rehiligha (generated)" id="1725484515185-68">
+        <addColumn tableName="two_windings_transformer_modification">
+            <column name="voltage_level_id2op" type="varchar(255)"/>
+        </addColumn>
+    </changeSet>
```
Comment (EtienneLt) [inline src/main/resources/db/changelog/changesets/changelog_20240904T211456Z.xml:202]: should it be in only one changeSet ?
maybe we should add a rollback ?

**Review by EtienneLt — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java (hunk 1 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java:130]: why ?

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java (hunk 3 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:57]: okey

File: src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java (hunk 1 above)
Comment (EtienneLt) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/modification/BranchModificationEntity.java:130]: well it is not synchronized with the attribute name but ok we do not change the schema

File: src/main/resources/db/changelog/changesets/changelog_20240904T211456Z.xml (hunk 4 above)
Comment (EtienneLt) [reply src/main/resources/db/changelog/changesets/changelog_20240904T211456Z.xml:202]: one changeSet for all the file

**Review by EtienneLt — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/AbstractBranchModification.java [hunk 5]
```diff
@@ -219,4 +223,11 @@ protected boolean characteristicsModified(BranchModificationInfos branchModifica
 
     protected abstract void modifyCharacteristics(Branch<?> branch, BranchModificationInfos branchModificationInfos,
             ReportNode subReportNode);
+
+    private ReportNode modifyBranchConnectivity1Attributes(BranchModificationInfos branchModificationInfos,
```
Comment (thangqp) [inline src/main/java/org/gridsuite/modification/server/modifications/AbstractBranchModification.java:227]:
```suggestion
    private ReportNode modifyBranchConnectivityAttributes(BranchModificationInfos branchModificationInfos,
```

**Review by thangqp — APPROVED** [review]

**Pull request merged**